File: pgm/ABC_Java/ABC_331_A.java
import java.io.*;
// import java.math.*;
// import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        final String[] MD = reader.readLine().split (" ");
        final String[] ymd = reader.readLine().split (" ");
        final int M = Integer.parseInt (MD[0]);
        final int D = Integer.parseInt (MD[1]);
        int y = Integer.parseInt (ymd[0]);
        int m = Integer.parseInt (ymd[1]);
        int d = Integer.parseInt (ymd[2]);
        d++;
        if (d > D) {
            d -= D;
            m++;
        }
        if (m > M) {
            m -= M;
            y++;
        }
        System.out.println (String.valueOf (y)
                                .concat (" ")
                                .concat (String.valueOf (m))
                                .concat (" ")
                                .concat (String.valueOf (d)));
        return;
    }
}

File: pgm/ABC_Kotlin/ABC_331_A.kt
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.lang.StringBuilder
// import java.time.*
// import java.util.StringTokenizer

// import kotlin.math.*

const val space: String = " "

fun main() {
    val (M: Int, D: Int) = readln().split(space).map { it.toInt() }
    var (y: Int, m: Int, d: Int) = readln().split(space).map { it.toInt() }
    d++
    if (d > D) {
        d -= D
        m++
    }
    if (m > M) {
        m -= M
        y++
    }
    println(arrayOf(y, m, d).joinToString(separator = space))
    kotlin.system.exitProcess(0)
}